lib/matplotlib/axis.py

```diff
@@ -932,17 +932,15 @@ def iter_ticks(self):
         """
         Iterate through all of the major and minor ticks.
         """
-        major_locs = self.major.locator()
-        major_ticks = self.get_major_ticks(len(major_locs))
+        major_locs = self.get_majorticklocs()
         self.major.formatter.set_locs(major_locs)
         major_labels = self.major.formatter.format_ticks(major_locs)
-
-        minor_locs = self.minor.locator()
-        minor_ticks = self.get_minor_ticks(len(minor_locs))
+        major_ticks = self.get_major_ticks(len(major_locs))
+        yield from zip(major_ticks, major_locs, major_labels)
```

Comment:
Can this get a comment?  I had to spend a few minutes understanding that `iter_ticks` is actually an iterator, and that `yield from` was returning iterator elements.  

I actually think this is very silly.  `iter_ticks` gets called at one spot (and one test) and gets made into a list right away.  So why not just return the list of tuples?

Reply:
Probably could be changed, but let's make this "future work"?  The use of an iterator dates back to 66290aa when ticking was factored out of `Axis.draw`, I agree it may as well return the list directly.  Needs deprecation, yadayada.
In the meantime, edited the method docstring.

Reply:
Done in #13363.